### PR TITLE
Fix duplicate messages and add info to render log

### PR DESCRIPTION
**Pull request by somewhatabstract**
The issue is that the `makeMiddleware` function in the `logging-watson` Google package adds its own transport if the second parameter does not indicate that one already exists. This results in two transports and therefore duplicate messages.

My first fix removed our own transport so that the middleware one gets added and is the only transport. This worked mostly except for anything that was logged outside of the express code. In those circumstances we got orphaned log messages and errors about a lack of a transport.

So, the fix I landed on was using our own transport as before, but also telling the middleware about it so that it does not add another. This addresses both the duplicate messages and the need for a transport outside of the express usage.

Unrelated, but low risk; this also adds the number of pending requests to the `/render` profiling just to give us some more date when we're looking at logs and traces.

## Testing
1. Deploy to GAE
2. Following our Confluence guide on testing the RRS, send some render requests and see that the logging is working, both for instance startup that logs outside of the express pipeline, and for subsequent requests inside the pipeline. Note that there are not duplicated entries in a request.
3. Update `webapp` to this commit SHA and verify with `make serve` that the dev experience works correctly still

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -94,7 +94,10 @@ app.use(
         const logging = getLogger(req);
 
         pendingRenderRequests++;
-        const renderProfile = profile.start(logging, "/render");
+        const renderProfile = profile.start(
+            logging,
+            `/render (active requests: ${pendingRenderRequests})`,
```
Comment (csilvers) [inline src/server.js]: (I take it this is unrelated?)

File: flow-typed/npm/winston_v3.x.x.js [hunk 2]
```diff
@@ -74,7 +74,7 @@ declare type $winstonLoggerConfig<T: $winstonLevels> = {
   format?: $winstonFormat,
   level?: $Keys<T>,
   levels?: T,
-  transports?: Array<$winstonTransport>,
+  transports?: $winstonTransport | Array<$winstonTransport>,
```
Comment (csilvers) [inline flow-typed/npm/winston_v3.x.x.js]: Are we allowed to do this?  Does winston actually allow a single transport object to be in `transports`?  I'm not sure why we would feel the need to do that.

File: src/logging.js [hunk 3]
```diff
@@ -34,45 +34,44 @@ function getFormatters() {
...
+    return new lw.LoggingWinston();
 }
 
 function initLogging(logLevel: LogLevel, isDev: boolean): Logger {
     // This is the logger that we use to log general information in our app.
     // Whereever one might use console, use this instead.
     const winstonLogger = winston.createLogger<NpmLogLevels>({
         level: logLevel,
-        transports: getTransports(isDev),
+        transports: getTransport(isDev),
```
Comment (csilvers) [inline src/logging.js]: Why not just say `transports: [getTransport(isDev)]` and keep the type signature the same as before?

Comment (somewhatabstract) [reply src/logging.js]: Oh, so the logger API actually accepts either (I had just been lazy with the flow types earlier). Seemed silly to create an array when we didn't need one.

File: flow-typed/npm/winston_v3.x.x.js (hunk 2 above)
Comment (somewhatabstract) [reply flow-typed/npm/winston_v3.x.x.js]: Yes, it does. I just had not updated the flow typing to match.

Comment (somewhatabstract) [reply flow-typed/npm/winston_v3.x.x.js]: Until now...

File: src/server.js (hunk 1 above)
Comment (somewhatabstract) [reply src/server.js]: Yes, it is. I was sneaking it in.

**Comment by somewhatabstract** [comment]
> The commit message could use some more detail saying exactly how this fixes things. I will be honest, it looks to me like there's no contentful change here at all. Or maybe changing from a list to a not-list is contentful somehow?

Good call. I'll update the comment right now. The fix is on the line we call `makeMiddleware` on the logging-watson stuff. That tells it that we have a transport so that it doesn't add its own.

I did it this way so that we had a transport for all our logging prior to that as well.

File: src/logging.js [hunk 4]
```diff
@@ -147,9 +146,11 @@ export function makeRequestMiddleware(logger: Logger): Promise<Middleware> {
               }),
           )
         : /**
-           * Otherwise, we're using the Google middleware
+           * Otherwise, we're using the Google middleware.
+           * This does some nice things including adding the appropriate
+           * logging transport for us.
            */
-          lw.express.makeMiddleware(logger);
+          lw.express.makeMiddleware(logger, getTransport(false));
```
Comment (csilvers) [inline src/logging.js]: Ah, ok, I see it now!

Comment (somewhatabstract) [reply src/logging.js]: > OK! I don't fully know the makeMiddleware syntax but this seems plausible.
> 
> But I don't understand why your code is so different from graphql-gateway. Of course, maybe graphql-gateway isn't working quite right, but I feel like it is. See what we do in services/graphql-gateway/src/main.js. Maybe you want to try the same thing here.

I think, based on looking at the graphql-gateway code, you actually do have this issue. You may just not have noticed the duplication.
